packages/nuxt/src/pages/runtime/plugins/prerender.ts

```diff
@@ -0,0 +1,40 @@
+import { defineNuxtPlugin } from "#app/nuxt";
+import { prerenderRoutes } from "#app/composables/ssr";
+// @ts-expect-error virtual file
+import _routes from "#build/routes";
+// @ts-expect-error virtual file
+import routerOptions from "#build/router.options";
+import type { RouterConfig } from "@nuxt/schema";
+import { joinURL } from "ufo";
+
+const OPTIONAL_PARAM_RE = /^\/?:.*(\?|\(\.\*\)\*)$/;
+
+export default defineNuxtPlugin(() => {
+  if (!import.meta.server || !import.meta.prerender) {
```

Comment:
I need to add guards for the condition below https://github.com/nuxt/nuxt/blob/5d071a70d3b8afdf450e62489c19ba22055480ae/packages/nuxt/src/pages/module.ts#L270

I'm still figuring out how to get all of these options available in a plugin, but wanted to throw up what I had in progress.

Reply:
A plugin is a great approach! Let me push a couple of changes, and I can think we can go ahead and merge 🙏